code/go/0chain.net/smartcontract/storagesc/config_test.go

```diff
@@ -0,0 +1,390 @@
+package storagesc
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func assertErrMsg(t *testing.T, err error, msg string) {
+	t.Helper()
+
+	if msg == "" {
+		assert.Nil(t, err)
+		return
+	}
+
+	if assert.NotNil(t, err) {
+		assert.Equal(t, msg, err.Error())
+	}
+}
+
+func Test_getConfig(t *testing.T) {
+	var (
+		ssc        = newTestStorageSC()
+		balances   = newTestBalances(t, false)
+		configured = setConfig(t, balances)
+		gn, _      = ssc.getConfig(balances, false)
+	)
+	assert.EqualValues(t, configured, gn)
+}
+
+func TestMinerSmartContractUpdate(t *testing.T) {
+	var (
+		ssc            = newTestStorageSC()
+		balances       = newTestBalances(t, false)
+		tx             = newTransaction(owner, ssc.ID, 0, 0)
+		originalConfig = setConfig(t, balances)
+		err            error
+		config         = new(scConfig)
+		update         = new(scConfig)
+	)
+
+	balances.txn = tx
+
+	// 1. Malformed update
+	t.Run("malformed update", func(t *testing.T) {
```

Comment:
For consistancy can now we adopt the [table driven subtest style](https://blog.golang.org/subtests#TOC_4.) of @MurashovVen's unit tests.